src/libraries/Microsoft.PowerFx.Core/Parser/TokenCursor.cs

```diff
@@ -103,11 +115,21 @@ public Token TokMove()
         {
             AssertValid();
             var tok = _currentToken;
+            
             if (_currentTokenId != TokKind.Eof)
             {
                 MoveTo(_currentTokenIndex + 1);
             }
 
+            var tokenLength = tok.Span.Lim - tok.Span.Min;
+            _currentCharIndex += tokenLength;
+
+            if (_flags.HasFlag(TexlParser.Flags.NamedFormulas))
```

Comment:
I don't think you need to base this off of the flag. You could add a function like `ResetOffset()` that does this and only call it when you move the offset in the outer parser.

Reply:
we still have to clone the spans for named formulas with new min & lim, I have tried creating them, if there is a difference in offset and min, but tests for Format() are failing as whitespaces created by format are ignored, will check with Bryan. please suggest if you have a different approach in mind.